.github/workflows/network-test.yaml

```diff
@@ -0,0 +1,158 @@
+name: "Network fault tolerance"
+
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        type: boolean
+        description: 'Run the build with tmate debugging enabled (https://github.com/marketplace/actions/debugging-with-tmate)'
+        required: false
+        default: false
+      wait_peer_disconnected:
+        type: boolean
+        description: 'Check other peers see the target peer is disconnected before running benchmark.'
+        required: false
+        default: false
+      target_peer:
+        description: 'Select the peer to target with netem package loss'
+        required: true
+        type: choice
+        options:
+          - alice
+          - bob
+          - carol
+        default: alice
```

Comment:
I think this will go away on a subsequent refactor of more advanced selection of the number of peers and which peers to target with the loss.

We can't use `--random` for two reasons: one, it blocks us a bit from loss on multiple peers; and it means we can't be clear about which traffic to which IP address to block.

Reply:
I definitely think peers_info_json.sh will go away in the subsequent PR that will make the peer configuration more generic/testable

I personally don't think it's worth refactoring away just at this moment; I think we can do it in a subsequent PR.

I agree that that then means that technically we could delete it (and do a bit more refactoring to support that); but my opinion is that it doesn't hurt to have it at the moment; it's more work to delete, and it indicates a nice idea that we want to replicate, even if not done in exactly the same way.

I.e. I'm certain we'll refactor it away subsequently.

Reply:
This suggestion has been pushed into a red-bin item on our board:
https://github.com/orgs/cardano-scaling/projects/6?pane=issue&itemId=77063030